lms/djangoapps/courseware/tests/test_discussion_xblock.py

```diff
@@ -243,3 +250,178 @@ def test_student_perms_are_correct(self, permissions):
+        """
+        self.client.login(username=self.user.username, password='test')
+        url = reverse('blocks_in_block_tree', kwargs={'usage_key_string': unicode(self.course_usage_key)})
+        query_params = {
+            'depth': 'all',
+            'username': self.user.username,
+            'block_types_filter': 'discussion',
+            'student_view_data': 'discussion'
+        }
+        response = self.client.get(url, query_params)
+        self.assertEquals(response.status_code, 200)
+        self.assertEquals(response.data['root'], unicode(self.course_usage_key))  # pylint: disable=no-member
+        for block_key_string, block_data in response.data['blocks'].iteritems():  # pylint: disable=no-member
+            block_key = deserialize_usage_key(block_key_string, self.course_key)
+            self.assertEquals(block_data['id'], block_key_string)
+            self.assertEquals(block_data['type'], block_key.block_type)
+            self.assertEquals(block_data['display_name'], self.store.get_item(block_key).display_name or '')
+            self.assertEqual(block_data['student_view_data'], {"topic_id": self.discussion_id})
+
+
+class TestXBlockQueryLoad(EnableTransformerRegistryMixin, SharedModuleStoreTestCase):
+    """
+    Test the number of queries executed when rendering the XBlock.
+    """
+
+    def test_permissions_query_load(self):
+        """
+        Tests that the permissions queries are cached when rendering numerous discussion XBlocks.
+        """
+        user = UserFactory.create()
+        course = ToyCourseFactory.create()
+        course_key = course.id
+        course_usage_key = self.store.make_course_usage_key(course_key)
+        discussions = []
+
+        for counter in range(5):
+            discussion_id = 'test_discussion_{}'.format(counter)
+            discussions.append(ItemFactory.create(
+                parent_location=course_usage_key,
+                category='discussion',
+                discussion_id=discussion_id,
+                discussion_category='Category discussion',
+                discussion_target='Target Discussion',
+            ))
+
+        # 3 queries are required to do first discussion xblock render:
+        # * django_comment_client_role
+        # * django_comment_client_permission
+        # * lms_xblock_xblockasidesconfig
+        num_queries = 3
+        for discussion in discussions:
+            discussion_xblock = get_module_for_descriptor_internal(
+                user=user,
+                descriptor=discussion,
+                student_data=mock.Mock(name='student_data'),
+                course_id=course.id,
+                track_function=mock.Mock(name='track_function'),
+                xqueue_callback_url_prefix=mock.Mock(name='xqueue_callback_url_prefix'),
+                request_token='request_token',
+            )
+            with self.assertNumQueries(num_queries):
```

Comment:
Thanks, @ormsbee , I'll leave it in.

@ormsbee and @robrap: Are you ready for me to squash and rebase to prepare for merge?